test/hack/cleanup/main.go

```diff
@@ -51,10 +56,22 @@ func main() {
 	ec2Client := ec2.NewFromConfig(cfg)
 	cloudFormationClient := cloudformation.NewFromConfig(cfg)
 	cloudWatchClient := cloudwatch.NewFromConfig(cfg)
+	iamClient := iam.NewFromConfig(cfg)
 
-	// Terminate any old instances that were provisioned by Karpenter as part of testing
-	// We execute these in serial since we will most likely get rate limited if we try to delete these too aggressively
-	ids := getOldInstances(ctx, ec2Client, expirationTime)
+	cleanupInstances(ctx, ec2Client, cloudWatchClient, expirationTime, logger)
```

Comment:
When I was thinking about parallelizing, I was thinking that you would just parallelize each of these individual functions so that the categories are parallelized. I think you actually save more time that way and you're less likely to get throttled from what I could think.